### ISSUE TITLE
[iOS] Could not load file or assembly 'Xamarin.iOS.Windows.Client, Version=1.0.0.0' A strongly-named assembly is required.

### ISSUE THREAD

**Issue by lukaszkn** [dotnet/maui#5629]
### Description

I've created new MAUI app from project template.
iOS target (local device) doesn't compile in Visual Studio 17.2.0 Preview 2.1 anymore with the following error below.
`Could not load file or assembly 'Xamarin.iOS.Windows.Client, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. A strongly-named assembly is required.`

**It was compiling and running fine in earlier preview versions.**

`1>C:\Program Files\dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.302-preview.14.122\tools\msbuild\iOS\Xamarin.iOS.HotRestart.targets(20,3): error MSB4018: The "HotRestart.Tasks.DetectSigningIdentity" task failed unexpectedly.`
`1>C:\Program Files\dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.302-preview.14.122\tools\msbuild\iOS\Xamarin.iOS.HotRestart.targets(20,3): error MSB4018: System.IO.FileLoadException: Could not load file or assembly 'Xamarin.iOS.Windows.Client, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. A strongly-named assembly is required. (Exception from HRESULT: 0x80131044)`
`1>C:\Program Files\dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.302-preview.14.122\tools\msbuild\iOS\Xamarin.iOS.HotRestart.targets(20,3): error MSB4018: File name: 'Xamarin.iOS.Windows.Client, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'`
`1>C:\Program Files\dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.302-preview.14.122\tools\msbuild\iOS\Xamarin.iOS.HotRestart.targets(20,3): error MSB4018:    at Xamarin.iOS.HotRestart.Tasks.DetectSigningIdentity.Execute()`
`1>C:\Program Files\dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.302-preview.14.122\tools\msbuild\iOS\Xamarin.iOS.HotRestart.targets(20,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()`
`1>C:\Program Files\dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.302-preview.14.122\tools\msbuild\iOS\Xamarin.iOS.HotRestart.targets(20,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()`
`1>C:\Program Files\dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.302-preview.14.122\tools\msbuild\iOS\Xamarin.iOS.HotRestart.targets(20,3): error MSB4018: `
`1>C:\Program Files\dotnet\packs\Microsoft.iOS.Windows.Sdk\15.2.302-preview.14.122\tools\msbuild\iOS\Xamarin.iOS.HotRestart.targets(20,3): error MSB4018:`
`1>Done building project "MauiApp2.csproj" -- FAILED.`
`========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========`

### Steps to Reproduce

1. Create new project -> .NET MAUI app
2. Choose target -> iOS local device / connected iPhone
3. Setup Hot restart if needed (Apple development team etc)
4. Try to compile / run the app

The code doesn't compile cause of the following:
```
error MSB4018: The "HotRestart.Tasks.DetectSigningIdentity" task failed unexpectedly.`
error MSB4018: System.IO.FileLoadException: Could not load file or assembly 'Xamarin.iOS.Windows.Client,
Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies.
A strongly-named assembly is required. (Exception from HRESULT: 0x80131044)`
```

### Version with bug

Preview 14 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 15

### Did you find any workaround?

NO

### Relevant log output

```shell
Microsoft Visual Studio Community 2022
Version 17.2.0 Preview 2.1
VisualStudio.17.Preview/17.2.0-pre.2.1+32317.152
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Community

ASP.NET and Web Tools 2019   17.2.240.24236
ASP.NET and Web Tools 2019

Azure App Service Tools v3.0.0   17.2.240.24236
Azure App Service Tools v3.0.0

C# Tools   4.2.0-2.22159.10+f3a5bad242b7a7b8149ae644de0a61c2f1bffc8d
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.6 (master@34d6af2)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Mono Debugging for Visual Studio   17.1.10 (65f98f5)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   6.2.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Razor (ASP.NET Core)   17.0.0.2215202+658e2db05d0daa9326356ee1edc85cd3d1b970c4
Provides languages services for ASP.NET Core Razor.

TypeScript Tools   17.0.10302.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.2.0-2.22159.10+f3a5bad242b7a7b8149ae644de0a61c2f1bffc8d
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.1.0-beta.22108.1+63bcc94efda4fe19b612b167a25fdbcee796040b
Microsoft Visual F# Tools

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

VSPackage Extension   1.0
VSPackage Visual Studio Extension Detailed Info

Xamarin   17.2.0.85 (d17-2-p2@c900809)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   17.2.0.64 (remotes/origin/main@3d19e6caf)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   17.1.13 (bb31b34)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   12.2.99.130 (main/e89ae42)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: f34bd77
    Java.Interop: xamarin/java.interop/main@aae23c97
    ProGuard: Guardsquare/proguard/v7.0.1@912d149
    SQLite: xamarin/sqlite/3.38.0@ccd83d8
    Xamarin.Android Tools: xamarin/xamarin-android-tools/main@f0b3abd


Xamarin.iOS and Xamarin.Mac SDK   15.7.0.410 (478c1d2c8)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

**Comment by Redth** [comment]
This is a known issue with Preview 14 and will be fixed in an upcoming VS preview release.

**Comment by RemcoDEV** [comment]
This bug unfortunately also slipped into VS2022 17.1.3 when building an iOS Xamarin project.

**Comment by StevenF42** [comment]
I have the same problem.  VS2022 17.1.3 building an iOS Xamarin project.

**Comment by adolfo1981** [comment]
> I have the same problem. VS2022 17.1.3 building an iOS Xamarin project.

Have you found a workaround? I ran into this after upgrading VS 2022 to version 17.1.3. Can't compile my XF solution :(

**Comment by StevenF42** [comment]
Unfortunately, the "workaround" is to uninstall VS 2022 and then install 17.1.2.  You can find the installer at:
https://docs.microsoft.com/en-us/visualstudio/releases/2022/release-history
I strongly suggest noting your configuration / extensions, etc. to make your life a bit easier after the install.

**Comment by adolfo1981** [comment]
Than you so much. I struggled to find the right installer to roll it back

**Comment by Extragorey** [comment]
@StevenF42 I see there are only downloads listed for Enterprise and Professional editions of VS2022. So Community users are just stuck?

**Comment by adolfo1981** [comment]
> @StevenF42 I see there are only downloads listed for Enterprise and Professional editions of VS2022. So Community users are just stuck?

Seems like 🙁. I realized that since I use community too. I spent hours trying to find the previous version and no luck. They force you to the latest for community.

**Comment by jfversluis** [comment]
Hey everyone, sorry this happened. We are aware this also happened for Xamarin and working to get it fixed. It should be fixed in the upcoming 17.2 preview 3 and hopefully we can get it in in time for 17.1.4, but it might be until 17.1.5.

For now you probably want to downgrade if you can.

**Comment by jfversluis** [comment]
Just got word that the fix was merged in 17.1.4 so the next preview should unblock you!

**Comment by RemcoDEV** [comment]
Thank you for your care Gerald!

**Comment by LucasCordova** [comment]
> Just got word that the fix was merged in 17.1.4 so the next release should unblock you!

This is blocking us too.  What is the approximate timeline / drop of 17.1.4? 

Thanks.

**Comment by alimassimi** [comment]
> Hey everyone, sorry this happened. We are aware this also happened for Xamarin and working to get it fixed. It should be fixed in the upcoming Visual Studio versions asap.
> 
> For now you probably want to downgrade if you can.

IF YOU CAN :) :) :), is this a challenge?!!!

**Comment by MMieszko** [comment]
Happened for me the same after update of VS 2022.

<img width="477" alt="image" src="https://user-images.githubusercontent.com/17722002/162924679-f3e26940-ba4d-4306-a182-fe7d150315cb.png">

**Comment by adolfo1981** [comment]
Looks like VS 2022 17.1.4 version is here folks. I'm downloading it now. Good luck! hope this fixes it

**Comment by inkysquid** [comment]
I can confirm that this problem is fixed in version 17.1.4.
